src/main/java/com/google/neighborgood/servlets/MessageServlet.java

```diff
@@ -131,6 +133,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     msgEntity.setProperty("sentTime", System.currentTimeMillis());
 
     datastore.put(msgEntity);
+
+    // After storing the message, create a notification entity to notify the receiver
+    Key taskKey = KeyFactory.stringToKey(taskId);
+
+    Entity taskEntity;
+    try {
+      taskEntity = datastore.get(taskKey);
+    } catch (EntityNotFoundException e) {
+      System.err.println("Unable to find the entity based on the input key");
```

Comment:
It could be useful to add information about the input key, e.g.
```
String.format("Unable to find the entity based on the input key %s for task %d", taskKey, taskId)
```

Reply:
Sure. I would change the error message in the new commit.